### PR TITLE
Use go-emoji to convert to html chars

### PR DESCRIPTION
**Pull request by hoffmabc**
Potential fix for #1542

### CODE REVIEW

File: core/core.go [hunk 1]
```diff
@@ -2,14 +2,19 @@ package core
...
 	"gx/ipfs/QmSY3nkMNLzh9GdbFKK5tT7YMfLpf52iUZ8ZRkr29MJaa5/go-libp2p-kad-dht"
 	libp2p "gx/ipfs/QmTW4SdgBWq9GjsBsHeUx8WuGxzhgzAf88UMH2w62PC8yK/go-libp2p-crypto"
 	ma "gx/ipfs/QmTZBfrPJmjWsCvHEtX5FE6KimVJhsJg5sBbqEFYf4UZtL/go-multiaddr"
-	cid "gx/ipfs/QmTbxNB1NwDesLmKTscr4udL2tVP7MaxvXnD1D9yX7g3PN/go-cid"
+	"gx/ipfs/QmTbxNB1NwDesLmKTscr4udL2tVP7MaxvXnD1D9yX7g3PN/go-cid"
 	"gx/ipfs/QmUadX5EcvrBmxAV9sE7wUWtWSqxns5K84qKJBixmcT1w9/go-datastore"
-	peer "gx/ipfs/QmYVXrKrKHDC9FobgmcmshCDyWwdrfwfanNQN4oxJ9Fk3h/go-libp2p-peer"
-	routing "gx/ipfs/QmYxUdYY9S6yg5tSPVin5GFTvtfsLauVcr7reHDD3dM8xf/go-libp2p-routing"
+	"gx/ipfs/QmYVXrKrKHDC9FobgmcmshCDyWwdrfwfanNQN4oxJ9Fk3h/go-libp2p-peer"
+	"gx/ipfs/QmYxUdYY9S6yg5tSPVin5GFTvtfsLauVcr7reHDD3dM8xf/go-libp2p-routing"
```
Comment (placer14) [inline core/core.go]: Can you undo the unrelated import changes?

File: core/core.go [hunk 2]
```diff
@@ -23,9 +28,8 @@ import (
 	"github.com/OpenBazaar/openbazaar-go/repo"
 	sto "github.com/OpenBazaar/openbazaar-go/storage"
 	"github.com/btcsuite/btcutil/hdkeychain"
-	"github.com/gosimple/slug"
 	"github.com/ipfs/go-ipfs/core"
-	logging "github.com/op/go-logging"
+	"github.com/op/go-logging"
```
Comment (placer14) [inline core/core.go]: Here, too.

File: core/core.go [hunk 3]
```diff
@@ -2,14 +2,19 @@ package core
 
 import (
 	"errors"
+	"fmt"
+	"regexp"
+	"unicode/utf8"
+
+	"github.com/gosimple/slug"
```
Comment (placer14) [inline core/core.go]: If the linter will let you, move this back where it was.

Comment (hoffmabc) [reply core/core.go]: this is new though.

**Comment by placer14** [comment]
I removed the dependency on the `assert` library. One less thing to vendor. Let's see if travis passes and then we'll merge.

**Comment by cpacia** [comment]
For this one maybe we should add a test to make sure non-emoji strings do not get mutated?
